Assets/Scripts/UI/Scoring.cs

```diff
@@ -2,23 +2,33 @@
 using System.Collections.Generic;
 using UnityEngine;
 using TMPro;
+using System;
 
 public class Scoring : MonoBehaviour
 {
     public TextMeshProUGUI Lives, Score,Gameover;
+    public event Action OnDeath;
```

Comment:
you should create all events in a separate service like I mentioned in the class - ServiceEvents.